### DISCUSSION TITLE
Не менять значение в ячейке, если не выполнено дейсивие

### DISCUSSION THREAD

**Discussion by f0gi** [totumonline/totum-ru-issues-and-discussions#587] [Как это сделать?] [answered]
Здравствуйте! Подскажите пожалуйста, возможно ли в поле тип - селект, при выборе определенного значения, открыть модальное окно с выбором значения, и после сохранения этого значения, меняется значение из селекта.

К примеру у меня есть некая запись, где поле селект, отвечает за статус: Новая, Завершена, На оплате
Если я меняю статус с Новая -> На оплате, то не менять значение у текущей записи, а вывести модальное окно, где я выбираю некий заказ и сохраняю, а после этого, только меняется значение текущего поля. Соответственно, если я закрываю модалку, то значение не меняется.

**Comment by erpic-ru** [comment, marked as answer]
Приветствую! Наверное, можно :) Я обычно на такие действия добавляю отдельные кнопки, которые ставят значение в селект и открывают нужные модалки. Шириной на 40 и только с иконкой в названии. Вроде как удобней получается + стараюсь делать, чтобы не статус управлял системой, а система статусом

**Reply by ttm-alexeypolunin** [reply]
Действия в Тотум срабатывают, когда значение изменилось. Те если изменили Заявка -> На оплате то сработал триггер. Нельзя поставить на паузу цепочку действий до совершения действия пользователем где-то в окне итп.

в этой логике что-то другое, например кнопка, должно инициировать действие открывающее окно из которого потом действием меняется статус. Кнопку можно засунуть в меню по правой кнопке мыши.

Если у вас комбинированный селект, те какие-то статусы ставяться руками, а какие-то через промежуточные действия, то можно блочить постановку некоторых статусов через процессную переменную https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/583

те ставите код действия который запускает errorExeption если переменная пустая

**Reply by f0gi** [reply]
@ttm-alexeypolunin могли бы показать простой пример, в поле по правому клику, кнопка действия, при нажатии создаёт запись в name_table?

**Reply by ttm-alexeypolunin** [reply]
Здесь про panel-функции: https://ru.docs.totum.online/specific-codes#panel-format

```
p1=: panelButton(code: $code; text: "Set value"; var: "table_name" = "some_table_name"; refresh: true)

    ```code:totum
    =: insert(table: $#table_name; field: 'some_field' = #id; log: true)
    ```
```

Функции панели получают при выполнении значения хедера/футера таблицы из которой запускаются и значения полей строки в которой запускаются, а также там работают `$#ntn`, `$#nci` и `$#nf` (но это не точно, смотрите в доках по ф.)

**Reply by ttm-alexeypolunin** [reply]
Еще я могу предположить такую логику: промежуточных техстатус, который пользователь ставит руками — ему запускается действие (окно/поле ввода/что угодно) и когда он его завершает мы ставим галочку например (но лучше дату со временем) в техническое поле - откуда проставляем полноценный статус.

Если он действие не завершил до конца, то потом обратно кроном откатываем в течении интервала времени (для этого и нужна была техдата со временем).

Но этот вариант более замороченный и обычно так сложно не нужно.
